logging/google/cloud/logging/handlers/app_engine.py

```diff
@@ -34,7 +34,7 @@
 _TRACE_ID_LABEL = 'appengine.googleapis.com/trace_id'
 
 
-class AppEngineHandler(CloudLoggingHandler):
+class AppEngineHandler(logging.StreamHandler):
```

Comment:
Can you explain why you changed the base class?

Reply:
Because after moving the code path of `get_gae_labels()` to `emit()`, we need to override the `emit()`  method. So it's fine either to extend from `CloudLoggingHandler` or `logging.StreamHandler`， as the `AppEngineHandler` now have all the methods in `CloudLoggingHandler`, I just extend from `logging.StreamHandler` directly.

Reply:
So CloudLoggingHandler is just for GKE now?  And/or GCE?  Can you add a comment to CloudLoggingHandler clarifying this?

Reply:
GKE has its own `ContainerEngineHandler`, in this [`get_default_handler`](https://github.com/GoogleCloudPlatform/google-cloud-python/blob/master/logging/google/cloud/logging/client.py#L295) method, the `CloudLoggingHandler` is returned if GAE and GKE environment are not detected.